python/lsst/daf/butler/registry/_defaults.py

```diff
@@ -120,6 +120,11 @@ def finish(self, registry: SqlRegistry) -> None:
             Raised if a non-governor dimension was included in ``**kwargs``
             at construction.
         """
+        # Skip re-initialization if it's already been completed.
+        # Can't just say 'self._finished' because this class is immutable.
+        if hasattr(self, "_finished"):
```

Comment:
Would it work to set
```
_finished: bool = False
```
at class scope instead (i.e. so it starts out as a ClassVar, and then gets shadowed)?

Reply:
Tried it but it didn't work unfortunately.